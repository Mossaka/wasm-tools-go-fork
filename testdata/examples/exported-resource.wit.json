{
  "worlds": [
    {
      "name": "exports",
      "imports": {},
      "exports": {
        "interface-0": {
          "interface": 0
        }
      },
      "package": 0
    }
  ],
  "interfaces": [
    {
      "name": "example",
      "types": {
        "example-resource": 0
      },
      "functions": {
        "[method]example-resource.foo": {
          "name": "[method]example-resource.foo",
          "kind": {
            "method": 0
          },
          "params": [
            {
              "name": "self",
              "type": 1
            }
          ],
          "results": []
        },
        "[method]example-resource.bar": {
          "name": "[method]example-resource.bar",
          "kind": {
            "method": 0
          },
          "params": [
            {
              "name": "self",
              "type": 1
            },
            {
              "name": "arg",
              "type": "u32"
            }
          ],
          "results": []
        }
      },
      "package": 0
    }
  ],
  "types": [
    {
      "name": "example-resource",
      "kind": "resource",
      "owner": {
        "interface": 0
      }
    },
    {
      "name": null,
      "kind": {
        "handle": {
          "borrow": 0
        }
      },
      "owner": null
    }
  ],
  "packages": [
    {
      "name": "examples:exported-resource@0.2.0",
      "interfaces": {
        "example": 0
      },
      "worlds": {
        "exports": 0
      }
    }
  ]
}