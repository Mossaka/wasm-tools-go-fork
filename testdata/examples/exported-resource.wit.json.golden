&wit.Resolve{
    Worlds: {
        &wit.World{
            Name:    "exports",
            Imports: {},
            Exports: {
                "interface-0": &wit.Interface{
                    Name:     &"example",
                    TypeDefs: {
                        "example-resource": &wit.TypeDef{
                            Name:       &"example-resource",
                            Kind:       &wit.Resource{},
                            Owner:      &wit.Interface{(CYCLIC REFERENCE)},
                            Docs:       wit.Docs{},
                            _worldItem: wit._worldItem{},
                            _type:      wit._type{},
                        },
                    },
                    Functions: {
                        "[method]example-resource.bar": &wit.Function{
                            Name: "[method]example-resource.bar",
                            Kind: &wit.Method{
                                Type:          &wit.TypeDef{(CYCLIC REFERENCE)},
                                _functionKind: wit._functionKind{},
                            },
                            Params: {
                                {
                                    Name: "self",
                                    Type: &wit.TypeDef{
                                        Name: (*string)(nil),
                                        Kind: &wit.BorrowedHandle{
                                            Type:    &wit.TypeDef{(CYCLIC REFERENCE)},
                                            _handle: wit._handle{},
                                        },
                                        Owner:      nil,
                                        Docs:       wit.Docs{},
                                        _worldItem: wit._worldItem{},
                                        _type:      wit._type{},
                                    },
                                },
                                {
                                    Name: "arg",
                                    Type: wit.U32{},
                                },
                            },
                            Results:    nil,
                            Docs:       wit.Docs{},
                            _worldItem: wit._worldItem{},
                        },
                        "[method]example-resource.foo": &wit.Function{
                            Name: "[method]example-resource.foo",
                            Kind: &wit.Method{
                                Type:          &wit.TypeDef{(CYCLIC REFERENCE)},
                                _functionKind: wit._functionKind{},
                            },
                            Params: {
                                {
                                    Name: "self",
                                    Type: &wit.TypeDef{
                                        Name: (*string)(nil),
                                        Kind: &wit.BorrowedHandle{
                                            Type:    &wit.TypeDef{(CYCLIC REFERENCE)},
                                            _handle: wit._handle{},
                                        },
                                        Owner:      nil,
                                        Docs:       wit.Docs{},
                                        _worldItem: wit._worldItem{},
                                        _type:      wit._type{},
                                    },
                                },
                            },
                            Results:    nil,
                            Docs:       wit.Docs{},
                            _worldItem: wit._worldItem{},
                        },
                    },
                    Package: &wit.Package{
                        Name: wit.Ident{
                            Namespace: "examples",
                            Package:   "exported-resource",
                            Extension: "",
                            Version:   &semver.Version{Major:0, Minor:2, Patch:0, PreRelease:"", Metadata:""},
                            _node:     wit._node{},
                        },
                        Interfaces: {
                            "example": &wit.Interface{(CYCLIC REFERENCE)},
                        },
                        Worlds: {
                            "exports": &wit.World{(CYCLIC REFERENCE)},
                        },
                        Docs: wit.Docs{},
                    },
                    Docs:       wit.Docs{},
                    _typeOwner: wit._typeOwner{},
                    _worldItem: wit._worldItem{},
                },
            },
            Package: &wit.Package{
                Name: wit.Ident{
                    Namespace: "examples",
                    Package:   "exported-resource",
                    Extension: "",
                    Version:   &semver.Version{Major:0, Minor:2, Patch:0, PreRelease:"", Metadata:""},
                    _node:     wit._node{},
                },
                Interfaces: {
                    "example": &wit.Interface{
                        Name:     &"example",
                        TypeDefs: {
                            "example-resource": &wit.TypeDef{
                                Name:       &"example-resource",
                                Kind:       &wit.Resource{},
                                Owner:      &wit.Interface{(CYCLIC REFERENCE)},
                                Docs:       wit.Docs{},
                                _worldItem: wit._worldItem{},
                                _type:      wit._type{},
                            },
                        },
                        Functions: {
                            "[method]example-resource.bar": &wit.Function{
                                Name: "[method]example-resource.bar",
                                Kind: &wit.Method{
                                    Type:          &wit.TypeDef{(CYCLIC REFERENCE)},
                                    _functionKind: wit._functionKind{},
                                },
                                Params: {
                                    {
                                        Name: "self",
                                        Type: &wit.TypeDef{
                                            Name: (*string)(nil),
                                            Kind: &wit.BorrowedHandle{
                                                Type:    &wit.TypeDef{(CYCLIC REFERENCE)},
                                                _handle: wit._handle{},
                                            },
                                            Owner:      nil,
                                            Docs:       wit.Docs{},
                                            _worldItem: wit._worldItem{},
                                            _type:      wit._type{},
                                        },
                                    },
                                    {
                                        Name: "arg",
                                        Type: wit.U32{},
                                    },
                                },
                                Results:    nil,
                                Docs:       wit.Docs{},
                                _worldItem: wit._worldItem{},
                            },
                            "[method]example-resource.foo": &wit.Function{
                                Name: "[method]example-resource.foo",
                                Kind: &wit.Method{
                                    Type:          &wit.TypeDef{(CYCLIC REFERENCE)},
                                    _functionKind: wit._functionKind{},
                                },
                                Params: {
                                    {
                                        Name: "self",
                                        Type: &wit.TypeDef{
                                            Name: (*string)(nil),
                                            Kind: &wit.BorrowedHandle{
                                                Type:    &wit.TypeDef{(CYCLIC REFERENCE)},
                                                _handle: wit._handle{},
                                            },
                                            Owner:      nil,
                                            Docs:       wit.Docs{},
                                            _worldItem: wit._worldItem{},
                                            _type:      wit._type{},
                                        },
                                    },
                                },
                                Results:    nil,
                                Docs:       wit.Docs{},
                                _worldItem: wit._worldItem{},
                            },
                        },
                        Package:    &wit.Package{(CYCLIC REFERENCE)},
                        Docs:       wit.Docs{},
                        _typeOwner: wit._typeOwner{},
                        _worldItem: wit._worldItem{},
                    },
                },
                Worlds: {
                    "exports": &wit.World{(CYCLIC REFERENCE)},
                },
                Docs: wit.Docs{},
            },
            Docs:       wit.Docs{},
            _typeOwner: wit._typeOwner{},
        },
    },
    Interfaces: {
        &wit.Interface{
            Name:     &"example",
            TypeDefs: {
                "example-resource": &wit.TypeDef{
                    Name:       &"example-resource",
                    Kind:       &wit.Resource{},
                    Owner:      &wit.Interface{(CYCLIC REFERENCE)},
                    Docs:       wit.Docs{},
                    _worldItem: wit._worldItem{},
                    _type:      wit._type{},
                },
            },
            Functions: {
                "[method]example-resource.bar": &wit.Function{
                    Name: "[method]example-resource.bar",
                    Kind: &wit.Method{
                        Type:          &wit.TypeDef{(CYCLIC REFERENCE)},
                        _functionKind: wit._functionKind{},
                    },
                    Params: {
                        {
                            Name: "self",
                            Type: &wit.TypeDef{
                                Name: (*string)(nil),
                                Kind: &wit.BorrowedHandle{
                                    Type:    &wit.TypeDef{(CYCLIC REFERENCE)},
                                    _handle: wit._handle{},
                                },
                                Owner:      nil,
                                Docs:       wit.Docs{},
                                _worldItem: wit._worldItem{},
                                _type:      wit._type{},
                            },
                        },
                        {
                            Name: "arg",
                            Type: wit.U32{},
                        },
                    },
                    Results:    nil,
                    Docs:       wit.Docs{},
                    _worldItem: wit._worldItem{},
                },
                "[method]example-resource.foo": &wit.Function{
                    Name: "[method]example-resource.foo",
                    Kind: &wit.Method{
                        Type:          &wit.TypeDef{(CYCLIC REFERENCE)},
                        _functionKind: wit._functionKind{},
                    },
                    Params: {
                        {
                            Name: "self",
                            Type: &wit.TypeDef{
                                Name: (*string)(nil),
                                Kind: &wit.BorrowedHandle{
                                    Type:    &wit.TypeDef{(CYCLIC REFERENCE)},
                                    _handle: wit._handle{},
                                },
                                Owner:      nil,
                                Docs:       wit.Docs{},
                                _worldItem: wit._worldItem{},
                                _type:      wit._type{},
                            },
                        },
                    },
                    Results:    nil,
                    Docs:       wit.Docs{},
                    _worldItem: wit._worldItem{},
                },
            },
            Package: &wit.Package{
                Name: wit.Ident{
                    Namespace: "examples",
                    Package:   "exported-resource",
                    Extension: "",
                    Version:   &semver.Version{Major:0, Minor:2, Patch:0, PreRelease:"", Metadata:""},
                    _node:     wit._node{},
                },
                Interfaces: {
                    "example": &wit.Interface{(CYCLIC REFERENCE)},
                },
                Worlds: {
                    "exports": &wit.World{(CYCLIC REFERENCE)},
                },
                Docs: wit.Docs{},
            },
            Docs:       wit.Docs{},
            _typeOwner: wit._typeOwner{},
            _worldItem: wit._worldItem{},
        },
    },
    TypeDefs: {
        &wit.TypeDef{
            Name:       &"example-resource",
            Kind:       &wit.Resource{},
            Owner:      &wit.Interface{(CYCLIC REFERENCE)},
            Docs:       wit.Docs{},
            _worldItem: wit._worldItem{},
            _type:      wit._type{},
        },
        &wit.TypeDef{
            Name: (*string)(nil),
            Kind: &wit.BorrowedHandle{
                Type:    &wit.TypeDef{(CYCLIC REFERENCE)},
                _handle: wit._handle{},
            },
            Owner:      nil,
            Docs:       wit.Docs{},
            _worldItem: wit._worldItem{},
            _type:      wit._type{},
        },
    },
    Packages: {
        &wit.Package{
            Name: wit.Ident{
                Namespace: "examples",
                Package:   "exported-resource",
                Extension: "",
                Version:   &semver.Version{Major:0, Minor:2, Patch:0, PreRelease:"", Metadata:""},
                _node:     wit._node{},
            },
            Interfaces: {
                "example": &wit.Interface{(CYCLIC REFERENCE)},
            },
            Worlds: {
                "exports": &wit.World{(CYCLIC REFERENCE)},
            },
            Docs: wit.Docs{},
        },
    },
}